### PR TITLE
[Feature #161278929] Add highlight  and comment functionality

### PR DESCRIPTION
**Pull request by asheux**
#### What does this PR do?
- Creates the functionality that enables users to highlight any part of a phrase on an article and be able to comment on the highlight

#### Description of Task to be completed?
- The main task that the backend is doing is saving the part of the text which has been highlighted, the user who is highlighting, the article itself and the comment a user posts for the highlight to the database.

#### How should this be manually tested?
```
$ git fetch origin pull/40/head:ft-highlight-and-comment-161278929
$ git checkout ft-highlight-and-comment-161278929
$ python manage.py makemigrations
$ python manage.py migrate
$ python manage.py runserver
```
#### Endpoints
```
- api/articles/<slug:slug>/highlight/ . [POST]
- api/articles/<slug:slug>/highlights/ . [GET, DELETE]
```
#### What are the relevant pivotal tracker stories?
[#161278929](https://www.pivotaltracker.com/n/projects/2205273/stories/161278929)

### CODE REVIEW

**Review by susanwere — APPROVED** [review]
Good work

**Review by botnetdobbs — APPROVED** [review]

**Review by gtsofa — COMMENTED** [review]
Nice implementation of the feature.

**Review by gtsofa — APPROVED** [review]

**Review by rickynyairo — APPROVED** [review]

File: authors/apps/highlight/serializers.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    """
+    Serializing the data received from the models
+    to display in dictionary format
+    """
+    highlighted = serializers.CharField(
+        required=True,
+        error_messages={"required": "Highlighted field is required"})
+    comment = serializers.CharField(
+        required=True,
+        error_messages={"required": "Comment filed is required"})
```
Comment (proxiex) [inline authors/apps/highlight/serializers.py]: Should this field be required? Can't a user just highlight a portion of an article? 🤷🏽‍♀️

Comment (asheux) [reply authors/apps/highlight/serializers.py]: No, that makes sense. It should not be required. Lemme fix that.

**Review by Adoniswalker — APPROVED** [review]

**Pull request merged**